lib/std/os/windows.zig

```diff
@@ -1091,6 +1166,43 @@ pub fn GetFinalPathNameByHandle(
     }
 }
 
+test "GetFinalPathNameByHandle" {
+    if (comptime builtin.os.tag != .windows)
+        return;
+    
+    const S = struct {fn check(file: std.fs.File, buffer: anytype, fmt: std.os.windows.GetFinalPathNameByHandleFormat) !usize {
+        const real = try std.os.windows.GetFinalPathNameByHandle(file.handle, fmt, buffer[0..]);
+        const offset = @divExact((@ptrToInt(real.ptr) - @ptrToInt(@ptrCast([*]u16, buffer))), 2);
+        return real.len + offset;
+    }};
+    
+    const file: std.fs.File = try std.fs.openSelfExe(.{}); //any file would do
+    defer file.close();
+    var buffer = std.mem.zeroes([1<<10]u16); //make this big enough for the test runner exe path
+
+    //check with sufficient size
+    const nt_length = try S.check(file, buffer[0..], .{.volume_name = .Nt});
+    const dos_length = try S.check(file, buffer[0..], .{.volume_name = .Dos});
+
+    //check with size insufficient by more than 1 character: always errors
+    std.testing.expect(null == S.check(file, buffer[0..nt_length-1], .{.volume_name = .Nt})
```

Comment:
```suggestion
    std.testing.expectEqual(null, S.check(file, buffer[0..nt_length-1], .{.volume_name = .Nt})
```

Ditto everywhere else, this way if the check fails it'll show you the faulty value.